baseapp/abci.go

```diff
@@ -176,6 +176,9 @@ func (app *BaseApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBeg
 		WithBlockGasMeter(gasMeter).
 		WithHeaderHash(req.Hash)
 
+	app.deliverState.ctx = app.deliverState.ctx.
+		WithConsensusParams(app.GetConsensusParams(app.deliverState.ctx))
```

Comment:
Do we still need https://github.com/cosmos/cosmos-sdk/blob/b3e922d08bb6daa042daad331d5cf5ae71b0c861/baseapp/baseapp.go#L491
?

Reply:
Do we need it for checkTx?

Reply:
added a condition to it.

Reply:
Just add `WithConsensusParams(app.GetConsensusParams(app.deliverState.ctx))` with the above `app.deliverState.ctx`. No need for a separate call.

Reply:
I was worried that the GetConsensusParams might get a uninitialized ctx when I write in that way. The gas meter and header hash not initialized.

Reply:
In that way, the ctx passed to GetConsensusParams is not initialized with gas meter and block header hash, is that a problem?

Reply:
> No, it's not a problem.

👍 changed.